src/extension/inspire/src/main/java/org/geoserver/inspire/wmts/InspireGridSetLoader.java

```diff
@@ -0,0 +1,55 @@
+/* (c) 2016 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.inspire.wmts;
+
+import org.geoserver.gwc.GWC;
+import org.geoserver.platform.ContextLoadedEvent;
+import org.geowebcache.grid.BoundingBox;
+import org.geowebcache.grid.GridSet;
+import org.geowebcache.grid.GridSetFactory;
+import org.geowebcache.grid.SRS;
+import org.springframework.context.ApplicationListener;
+
+/**
+ * Loads the inspire grid set and mark it as non editable by the user.
+ */
+public class InspireGridSetLoader implements ApplicationListener<ContextLoadedEvent> {
```

Comment:
If the grid set is not present it will be added and consequently stored in ``geowebcache.xml``. If a data dircetory coming from a GeoServer that has the INSPIRE extension enabled is used in a a GeoServer that doesn't have the INSPIRE extension enabled nothing will be done to the grid set but the user will be able to edit it or remove it from GeoServer UI.